DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/Factories/DicomProcessorFactory.cs

```diff
@@ -22,8 +26,23 @@ public IAnonymizerProcessor CreateProcessor(string method, JObject settingObject
                 "remove" => new RemoveProcessor(),
                 "refreshuid" => new RefreshUIDProcessor(),
                 "keep" => new KeepProcessor(),
-                _ => null,
+                _ => CreateCustomProcessor(method)
             };
         }
+
+        public void AddCustomerProcessor(string method, IAnonymizerProcessor processor)
+        {
+            EnsureArg.IsNotNullOrEmpty(method, nameof(method));
+            EnsureArg.IsNotNull(processor, nameof(processor));
+
+            _customerProcessors[method.ToLower()] = processor;
```

Comment:
Do we need to check duplicates here? For example, can customers create/override a custom processor for our built-in method `redact`? For another example, can customers call this function twice with the same method but different processors?

Reply:
For now, 
Customers can override build-in processors or customized processors by adding twice.
Do we need to forbid overriding build-in processors? Or re-design the duplicate behaviors?

Reply:
For FHIR, the current behavior is that the built-in processors could be overridden by customized processors, as they shared the same `_processors`. For DICOM, the customized processors are kept separately. Can they be overridden in current design?

Reply:
I think it would be better to add some checks to prevent overriding built-in processors.

Reply:
Agree, I feel overriding built-in processors is not desired. It will be confusing to diagnose the de-id output. Customers can use their own method name for customized actions like `myreadct`, `mydateshift`

Reply:
Sure, I added a check to prevent override built-in processor.